android/src/main/kotlin/com/rohit/ble_peripheral/BlePeripheralPlugin.kt

```diff
@@ -173,8 +173,6 @@ class BlePeripheralPlugin : FlutterPlugin, BlePeripheralChannel, ActivityAware,
     private fun validatePermission(): Boolean {
         val permissionsList = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
             arrayOf(
-                Manifest.permission.BLUETOOTH_ADMIN,
-                Manifest.permission.BLUETOOTH,
                 Manifest.permission.BLUETOOTH_SCAN,
```

Comment:
@alisherbu are you sure we don't need these permissions ?
also i haven't tested if we really need BluetoothScan permission

Reply:
Yeap, these are legacy permissions for older devices. https://developer.android.com/guide/topics/connectivity/bluetooth/permissions

Reply:
I think so. I will tes BluetoothScan permission later

Reply:
@alisherbu i would suggest to revert the Permission changes, i was also planning to recheck the permissions later
And we can merge rest of the changes for now

Reply:
When I validate Manifest.permission.BLUETOOTH, OS(Samsung Android 12) always returns false (not granted). So I had to remove these

Reply:
@alisherbu for now you can add all these permissions like in example app, and we can work with the permissions in new pr

Reply:
On my google pixel device this check also fails every time even if all permissions are granted.

Reply:
@Bennik2000 have you tried example app on main branch

Reply:
The example indeed works and the device gets discovered. However I get the following log `Need BLUETOOTH PRIVILEGED permission: Neither user 10704 nor current process has android.permission.BLUETOOTH_PRIVILEGED` but I guess this is a different issue.

Reply:
@Bennik2000 Seems like you are also facing [this](https://github.com/rohitsangwan01/ble_peripheral/issues/1#issuecomment-1709635050) issue with the permissions